### ISSUE TITLE
[BUG] All volumes degraded at the same time

### ISSUE THREAD

**Issue by Hr46ph** [longhorn/longhorn#10091]
## Describe the bug

Without apparent reason, all volumes are suddenly degraded. Some with one failed replica, others with 2. Several pods are left without data locality.

## To Reproduce

I have no idea how to replicate, it just happened (not for the first time I might add).

## Expected behavior

Reliability. This stuff scares me.

## Support bundle for troubleshooting

WIll add after posting as I need a ticket number.

## Environment

 - Longhorn version: 1.7.2
 - Impacted volume (PV): All of them
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: Talos Linux 1.8.0
   - Number of control plane nodes in the cluster: 3
   - Number of worker nodes in the cluster: 6 including control plane nodes
 - Node config
   - OS type and version: talos 1.8.0
   - Kernel version: 6.6.60-talos
   - CPU per node: 6
   - Memory per node: 64GB
   - Disk type (e.g. SSD/NVMe/HDD): NVMe
   - Network bandwidth between the nodes (Gbps): 1Gbps
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal
 - Number of Longhorn volumes in the cluster: 19

## Additional context

I restarted a postgres pod, went to check its actual size in Longhorn GUI and all volumes went degraded all at once.

## Workaround and Mitigation

Patience, I hope. Its still recovering.

**Comment by derekbit** [comment]
Can you provide a support bundle?

**Comment by Hr46ph** [comment]
> Can you provide a support bundle?

Yes, I just did. I wanted to complete the bug form first so I could provide the github bug report link with the support bundle. I sent it to longhorn-support-bundle at suse dot com.

**Comment by derekbit** [comment]
One quick question, are the failed replica on the same node?

**Comment by Hr46ph** [comment]
> One quick question, are the failed replica on the same node?

No, they were all over the place. Mostly on the wk01, 02 and 03 (I assume you see hostnames in the support bundle).

It did recover fairly quickly, only one volume is rebuilding. Maybe I panicked too quickly but still I don't think this should happen so I really hope you can see why this happens.

Thanks for the quick responses.

**Comment by derekbit** [comment]
> Network bandwidth between the nodes (Gbps): 1Gbps

BTW, the network bandwidth is not suitable for Longhorn, and 10 Gbps NIC is recommended. Please check the best practice https://longhorn.io/docs/1.7.2/best-practices/#volume-performance-optimization.

**Comment by Hr46ph** [comment]
> > Network bandwidth between the nodes (Gbps): 1Gbps
> 
> BTW, the network bandwidth is not suitable for Longhorn, and 10 Gbps NIC is recommended. Please check the best practice https://longhorn.io/docs/1.7.2/best-practices/#volume-performance-optimization.

It's not a production cluster its one of the compromises to keep costs to a minimum. Its also 1 disk which not recommended. There will be dedicated disks added in the near future though.

**Comment by derekbit** [comment]
@Hr46ph Has the support bundle provided?

**Comment by Hr46ph** [comment]
> @Hr46ph Has the support bundle provided?

Yes, I did so when I replied [yesterday](https://github.com/longhorn/longhorn/issues/10091#issuecomment-2564283903).

**Comment by derekbit** [comment]
Thanks @Hr46ph. 
I've checked the support bundle. There are lots of I/O timeout errors. Besides Longhorn, other services were experiencing errors. I think it is a temporary network issue in the environment.

**Comment by Hr46ph** [comment]
> Thanks @Hr46ph. I've checked the support bundle. There are lots of I/O timeout errors. Besides Longhorn, other services were experiencing errors. I think it is a temporary network issue in the environment.

In which log do you see those so that I can look into those further. Network is professional equipment with certified cabling. In network management softtware I dont see errors (whcih doesnt mean there arent other issues).

I can grep -ri over the extracted  supportbundle I see many hits on timeouts, except most are harmless.

If you see anything specific can you share that please so I can look into that.

Thanks!

**Comment by derekbit** [comment]
@Hr46ph 

You can check the messages
`... level=error msg="R/W Timeout. No response received in 8s" func="dataconn.(*Client).loop" file="client.go:168"...`.
The latencies between engines to the downstream replicas exceeded 8 seconds and lead to failed replicas. 

Were there any I/O spikes during this period?

**Comment by innobead** [comment]
For this kind of low-spec/non-prod env, might tune the timeout via https://longhorn.io/docs/1.7.2/references/settings/#engine-replica-timeout.

**Comment by derekbit** [comment]
@Hr46ph 

I cannot find any clues related to the termination of the `longhorn-manager-kg4vl` pod. I also found the kubelet.log and k3s log are all empty. cc @c3y1huang Do you know the root cause?

`longhorn-manager-kg4vl` is a control plane pod, so the termination won't impact the data plane's IO.
You mention the high CPU usage due to the snapshot checksum hashing. I'm thinking if the full story is 
```
snapshot checksum hashing (background IO) -> disk is busy -> disk is unable to serve requests from applications (foreground IO) within 8 seconds-> lead to failed IO
```

For the CPU usage caused by snapshot checksum hashing, we will improve the background IO throttling. 
If the issue is triggered again, you can try to disable snapshot checksum hashing and see if the issue remains.

**Comment by derekbit** [comment]
@Hr46ph 

I cannot find any clues related to the termination of the `longhorn-manager-kg4vl` pod. I also found the kubelet.log and k3s log are all empty. cc @c3y1huang Do you know the root cause?

`longhorn-manager-kg4vl` is a control plane pod, so the termination won't impact the data plane's IO.
You mention the high CPU usage due to the snapshot checksum hashing. I'm thinking if the full story is 
```
snapshot checksum -> disk is busy -> disk cannot serve foreground IO from applications -> lead to failed IO
```

For the CPU usage caused by snapshot checksum hashing, we will improve the background IO throttling.

**Comment by derekbit** [comment]
Let's keep the ticket opened for
- the empty log issue (cc @c3y1huang)
- waiting for @Hr46ph's feedback

**Comment by Hr46ph** [comment]
@derekbit Quick update I've been monitoring the logs for timeouts and other issues and so far its been quiet. I've not had issues so far with volumes spontaneously degrading.
